### PR TITLE
helm unittest

### PR DESCRIPTION
**Pull request by jwtodd**
make the helm unit tests green so that we can more readily develop-and-deploy non-breaking kubernetes changes

test:

```
% ./bin/mdsctl test:unit
```

### CODE REVIEW

**Comment by macsj200** [comment]
I'm not terribly familiar with helm/its testing strategies, what's the tl;dr for how these tests work?

File: helm/mds/tests/auth-metrics-jwt-audience_test.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+suite: test authetication jwt audience metrics
```
Comment (janedotx) [inline helm/mds/tests/auth-metrics-jwt-audience_test.yaml]: nit but you might wanna rename to "authentication"

**Comment by janedotx** [comment]
I just get a bunch of help text when I run `bin/mdsctl test:unit`

File: helm/mds/tests/auth-metrics-jwt-audience_test.yaml (hunk 1 above)
Comment (jwtodd) [reply helm/mds/tests/auth-metrics-jwt-audience_test.yaml]: appreciate the 👀 @janedotx :) thx!

**Comment by jwtodd** [comment]
@janedotx did you clone this branch?

might need to `./bin/mdsctl bootstrap` to seed the requisite toolchain (eg helm-unittest, diff, secrets)

`./bin/mdsctl install:helm` is a short circuit for the helm toolchain install, ie a subset of bootstrap

perhaps we should tandem it up

**Comment by jwtodd** [comment]
@janedotx a run should look something like the following:

➜  mds-core git:(feature/jwtodd-helm-unittest) ./bin/mdsctl test:unit                                                           (feature/jwtodd-helm-unittest)mds-core

### Chart [ mds ] /Users/jwtodd/projects/mds-core/helm/mds

 PASS  test cronjob provider-processor  helm/mds/tests/cronjob-provider-processor_test.yaml
 PASS  test deployment geography-author helm/mds/tests/deployment-geography-author_test.yaml
 PASS  test egress nats helm/mds/tests/egress-nats_test.yaml
 PASS  test authetication config jwt audience   helm/mds/tests/auth-config-jwt-audience_test.yaml
 PASS  test authetication native jwt audience   helm/mds/tests/auth-native-jwt-audience_test.yaml
 PASS  test autoscaler daily    helm/mds/tests/autoscaler-daily_test.yaml
 PASS  test autoscaler native   helm/mds/tests/autoscaler-native_test.yaml

**Comment by jwtodd** [comment]
@macsj200 helm-unittest is a utility that expresses helm test assertions as yaml files that mirror that of the target helm charts. it reads relatively fluent has has a relatively rich collection of assertion statements. i personally find helm-unittest a nice helm dev tool alongside helm-template, which effectively materializes the template with the associated context overlays consistent with the deployment helm-install command without actually executing the commands. as such, helm-unittest is a means to persist the assertions in such a way there the integrity of the helm charts can readily be verified, not unlike unit-testing in general. to that point, in envision our build automation pipeline to invoke helm unit tests alongside code unit tests in kind.

**Comment by jwtodd** [comment]
@macsj200 here is another helm unit-test framework, but i appreciate the declarative approach of the helm unit-test plugin as opposed to compiling test code:

https://blog.gruntwork.io/automated-testing-for-kubernetes-and-helm-charts-using-terratest-a4ddc4e67344

**Comment by jwtodd** [comment]
@janedotx added a pre-check to ensure a resolvable k8s endpoint is available or else we will emit a contextual error message. thx for the suggestion.

**Pull request merged**